ui/src/map/map-view.tsx

```diff
@@ -356,27 +366,24 @@ const MapView: React.FC<MapViewProps> = ({
             onHoverLocation: props.onHoverLocation,
             onClickLocation: props.onClickLocation,
         };
-        const deactivateToolFunc = selectToolBasic.activate(olMap, layerAdapters, toolActivateOptions);
-
-        // Always activate e.g. highlight tool
-        const deactivateHighlightTool = highlightTool.activate(
-            olMap,
-            layerAdapters,
-            toolActivateOptions,
-        );
-        const deactivatePointLocationTool = pointLocationTool.activate(
-            olMap,
-            layerAdapters,
-            toolActivateOptions,
-        );
+
+        const deactiveToolFunctions = [
+            pointLocationTool.activate(olMap, layerAdapters, toolActivateOptions),
+        ];
+
+        if (!measurementToolActive) {
+            deactiveToolFunctions.push(
+                selectToolBasic.activate(olMap, layerAdapters, toolActivateOptions),
+            );
+
+            deactiveToolFunctions.push(
+                highlightTool.activate(olMap, layerAdapters, toolActivateOptions),
+            );
+        }
```

Comment:
En ihan ymmärtänyt tätä kohtaa 🤔 Lähinnä siis siltä osalta, että meillä on muuttujan nimen mukaan kokoelma `deactivate`-funktioita, mutta sinne kumminkin tungetaan aina `activate`-kutsuja, ja myös `highlightTool`:in aktivointi riippuu `measurementTool`:ista

Reply:
Joo, tämä hämmensiä itteänikin. Activate-funkkari palauttaa funkkarin, jolla siistitään työkalun event-bindingit eli siistitään vanhat listenerit pois.
Ja if-ehto sen takia, että kun mitataan niin se ei saisi silloin valita alla olevaa raidetta